examples/gno.land/p/demo/uuid/uuid.gno

```diff
@@ -0,0 +1,125 @@
+// Package uuid provides utility functions for generating unique identifiers
+// (UUIDs) based on a custom implementation inspired by the Snowflake algorithm (https://pkg.go.dev/github.com/godruoyi/go-snowflake#section-sourcefiles).
+// The UUIDs generated by this package ensure uniqueness and are suitable for
+// distributed systems. The package offers a mechanism to convert these IDs
+// to standard UUID format strings, making them compatible with systems
+// requiring UUIDs.
+// Add import "gno.land/p/demo/entropy" to use the entropy package. For of the pseudo-random number generator.
+// TimesTampLength is 32 because entropy.Value() returns a uint32.
+// 32 it's the maximum value that can be returned "4 294 967 295", It's represent 136 years (4 294 967 295 secondes)
+
+package uuid
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"std"
+
+	"gno.land/p/demo/entropy"
+	"gno.land/p/demo/ufmt"
+)
+
+const (
+	TimestampLength uint8  = 32
+	MachineIDLength uint64 = 10
+	SequenceLength  uint64 = 12
+	MaxSequence     uint16 = 1<<SequenceLength - 1
+	MaxTimestamp    uint32 = 1<<TimestampLength - 1
+	MaxMachineID    uint64 = 1<<MachineIDLength - 1
+
+	machineIDMoveLength = SequenceLength
+	timestampMoveLength = MachineIDLength + SequenceLength
+
+	startTime uint32 = 1226354400 // 10 Nov 2008 23:00:00 UTC in seconds (Snowflake epoch)
```

Comment:
is this for some kind of compatibility, or could it just be 1/1/24?

i see no reason to use epochs from other formats unless it's to ensure some kind of sequencing / compatiblity from existing IDs.